tests/zkevm/test_worst_compute.py

```diff
@@ -515,6 +515,86 @@ def jump_seq():
     )
 
 
+@pytest.mark.zkevm
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.slow
+def test_worst_jumpi_fallthrough(
+    state_test: StateTestFiller,
+    pre: Alloc,
+):
+    """Test running a JUMPI-intensive contract with fallthrough."""
+    env = Environment()
+
+    def jumpi_seq():
+        return Op.JUMPI(Op.PUSH0, Op.PUSH0)
+
+    bytes_per_seq = len(jumpi_seq())
+    seqs_per_call = MAX_CODE_SIZE // bytes_per_seq
+
+    # Create and deploy the jumpi-intensive contract
+    jumpis_code = jumpi_seq() * seqs_per_call
+    jumpis_address = pre.deploy_contract(code=bytes(jumpis_code))
+
+    # Call the contract repeatedly until gas runs out.
+    caller_code = While(body=Op.POP(Op.CALL(address=jumpis_address)))
```

Comment:
For this test, the `CALL` should be removed and a while loop should be used for the gas optimization.